src/mlpack/methods/ann/layer/atrous_convolution_impl.hpp

```diff
@@ -37,6 +37,39 @@ AtrousConvolution<
   // Nothing to do here.
 }
 
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+AtrousConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::AtrousConvolution(
+  const AtrousConvolution& layer) :
+    inSize(layer.inSize),
+    outSize(layer.outSize),
+    kernelWidth(layer.kernelWidth),
+    kernelHeight(layer.kernelHeight),
+    weights(layer.weights),
+    strideWidth(layer.strideWidth),
+    strideHeight(layer.strideHeight),
+    inputWidth(layer.inputWidth),
+    inputHeight(layer.inputHeight),
+    outputWidth(layer.outputWidth),
+    outputHeight(layer.outputHeight),
+    dilationWidth(layer.dilationWidth),
+    dilationHeight(layer.dilationHeight),
+    padding(layer.padding)
+{
+  Reset();
```

Comment:
It seems like this implementation doesn't copy only the things that are serialized in `serialize()`, and I don't think that the `Reset()` call is needed, since when layers are copied by `FFN` (or another network type), `Reset()` will be called by `FFN`'s copy constructor.  Maybe you haven't changed this one yet to match that pattern, or maybe I overlooked something?  Sorry if my comment is out of place. :+1: